Switch to Rocky 9 built external packages on Hercules

WalterKolczynski-NOAA opened the pull request:
# Description
The workflow was updated to use modules built on Rocky 9, but the external packages (like prepobs) were still pointing to the versions built on CentOS (Orion). This transitions to packages built on Rocky 9.

Updating of the tracker package has been deferred until later. As such, the tracker jobs have been disabled by returning immediately if they are on Hercules. Since these jobs are small, resource-wise, it should not meaningfully impact turnover time.

# Type of change
- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? YES
  - Tracker has not yet been updated to build on Hercules
- Does this change require a documentation update? NO

# How has this been tested?
- Cycled test on Hercules

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

WalterKolczynski-NOAA committed:
Switch to Rocky 9 built external packages on Hercules

The workflow was updated to use modules built on Rocky 9, but the
external packages (like prepobs) were still pointing to the versions
built on CentOS (Orion). This transitions to packages built on Rocky
9.

WalterKolczynski-NOAA commented:
The only update needed to the packages was a minor change to the Fit2Obs spack-stack module path:
```diff
-prepend_path("MODULEPATH", "/work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/envs/unified-env/install/modulefiles/Core")
+prepend_path("MODULEPATH", "/work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/envs/gsi-addon-env/install/modulefiles/Core")
```

Changes are made in-place without changing the version number of the directory (yet).

DavidHuber-NOAA commented:
It looks like the TC_Tracker is not in the specified location:
```console
hercules-login-1[3] ~$ ls /work/noaa/global/glopara/git_rocky9/TC_tracker/
hercules-login-1[4] ~$
```

WalterKolczynski-NOAA commented:
> It looks like the TC_Tracker is not in the specified location:
> 
> ```
> hercules-login-1[3] ~$ ls /work/noaa/global/glopara/git_rocky9/TC_tracker/
> hercules-login-1[4] ~$
> ```

Yeah, this was a known thing. I thought I had tested this before I launched CI, but might've mixed it up with another test I did over the weekend.

WalterKolczynski-NOAA committed:
Disable tracker jobs on Hercules

The external tracker package is not yet available on Rocky 9, so a
hack is added to skip those jobs on Hercules until the package is
available. This will need to be extended to Orion if that machine
transitions to Rocky 9 before a package is available.

Refs #2639

WalterKolczynski-NOAA committed:
Relocate hack to skip tracker on Hercules

Checking of the machine had to be moved after config.base is sourced
as that is where the machine is set.

WalterKolczynski-NOAA committed:
Merge branch 'develop' into feature/migrate_external_rocky9

WalterKolczynski-NOAA committed:
Use correct case for machine name

WalterKolczynski-NOAA committed:
Merge branch 'develop' into feature/migrate_external_rocky9
